velox/vector/BaseVector.cpp

```diff
@@ -551,10 +551,22 @@ VectorPtr newConstant(
     velox::memory::MemoryPool* pool) {
   using T = typename KindToFlatVector<kind>::WrapperType;
   T copy = T();
-  if (!value.isNull()) {
-    if constexpr (std::is_same_v<T, StringView>) {
+  TypePtr type;
+  if constexpr (std::is_same_v<T, StringView>) {
+    type = Type::create<kind>();
+    if (!value.isNull()) {
       copy = StringView(value.value<kind>());
-    } else {
+    }
+  } else if constexpr (
+      std::is_same_v<T, ShortDecimal> || std::is_same_v<T, LongDecimal>) {
```

Comment:
It would be nice to extract this change in a separate PR and add a test.

Reply:
I will make this change.